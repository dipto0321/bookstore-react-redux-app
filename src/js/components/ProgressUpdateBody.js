import React from 'react';
import PropTypes from 'prop-types';
import {connect } from 'react-redux';
import { updatePage} from '../actions/book';

export class ProgressUpdateBody extends React.Component {
  state = {
    current: this.props.current_page || "0",
  }

  handleChange = (target) => {
    const { book_id, updatePage } = this.props;
    this.setState({
      [target.name]: target.value,
    });

    updatePage(book_id, target.value);
  }

  render(){
    const { pagesArray, id, title } = this.props;
    const { current} = this.state;
    return (
      <div>
        <h1>{title}</h1>
        <strong>Now reading page: </strong>
        <select
          name="current"
          value={current}
          onChange={e => this.handleChange(e.target)}
        >
          {
            ['Not started', ...pagesArray].map(page => (
              <option
                value={page}
                key={`${id}-${page}
                `}
                id={`${id} ${page}
                `}
              >
                {page}
              </option>
            ))
          }
        </select>
      </div>
    )
  }
}

ProgressUpdateBody.propTypes = {
  pagesArray: PropTypes.instanceOf(Object).isRequired,
  book_id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  updatePage: PropTypes.func.isRequired,
};

export default connect(null, { updatePage})(ProgressUpdateBody);
