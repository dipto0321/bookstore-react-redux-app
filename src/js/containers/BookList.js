import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { switchAddBookMode, removeBook, switchProgressUpdate } from '../actions/books';
import { changeFilter } from '../actions/filter';
import Book from '../components/Book';
import CategoryFilter from '../components/CategoryFilter';
import ProgressUpdateModal from '../components/ProgressUpdateModal';

export const BookList = (
  {
    books,
    changeFilter,
    filter,
    removeBook,
    switchAddBookMode,
    progressUpdateMode,
    switchProgressUpdate,
  },
) => (
  <div>
    <ProgressUpdateModal
      progressUpdateMode={progressUpdateMode}
      switchProgressUpdate={switchProgressUpdate}
    />
    <div className="booklist-controls">
      <CategoryFilter
        filter={filter}
        handleChange={optionValue => changeFilter(optionValue)}
      />
      <button
        className="add-book-btn btn-sm"
        type="button"
        onClick={switchAddBookMode}
      >
          Add Book
      </button>
    </div>
    <div>
      {books.map(book => (
        <Book
          key={book.id}
          book={book}
          handleRemove={id => removeBook(id)}
        />
      ))}
    </div>
  </div>
);

const mapStateToProps = state => ({
  books: state.books.filter(book => (state.filter === '' ? true : book.category === state.filter)),
  filter: state.filter,
  progressUpdateMode: state.progressUpdateMode,
});

BookList.propTypes = {
  books: PropTypes.instanceOf(Object).isRequired,
  changeFilter: PropTypes.func.isRequired,
  filter: PropTypes.string.isRequired,
  removeBook: PropTypes.func.isRequired,
  switchAddBookMode: PropTypes.func.isRequired,
  progressUpdateMode: PropTypes.instanceOf(Object).isRequired,
  switchProgressUpdate: PropTypes.func.isRequired,
};

export default connect(
  mapStateToProps,
  {
    removeBook, changeFilter, switchAddBookMode, switchProgressUpdate,
  },
)(BookList);
