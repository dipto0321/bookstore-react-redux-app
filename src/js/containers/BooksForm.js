import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import categories from '../data/bookCategories';

const BooksForm = () => {
  return (
    <form>
      <input type="text"/>
      <select>
        {
          categories.map(category => (
            <option
              key={category}
              value={category}
            >
              { category }
            </option>
          ))
        }
      </select>
      <button 
        type="submit"      
      >
        Add Book
      </button>
    </form>
  )
};

export default BooksForm;
