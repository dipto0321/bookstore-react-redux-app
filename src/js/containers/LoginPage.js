import React, { useState, useEffect } from 'react';
import { connect } from 'react-redux';
import { login } from '../thunks/currentUser';

export const LoginPage = ({ login, error, history }) => {
  const [emailOrUsername, setEmailOrUsername] = useState('');
  const [password, setPassword] = useState('');
  const [errorMessage, setErrorMessage] = useState(error);

  useEffect(() => {
    setErrorMessage(error);
  }, [error]);

  const loginUser = (e) => {
    e.preventDefault();
    login({ email_or_username: emailOrUsername, password })
      .then(() => {
        setEmailOrUsername('');
        setPassword('');
        setErrorMessage(error);
        history.push('/');
      }).catch(() => {});
  };

  return (
    <form>
      {
        error && <div className="error">{errorMessage}</div>
      }
      <div>
        <label htmlFor="email">Username/Email:</label>
        <input
          type="email"
          id="email"
          value={emailOrUsername}
          onChange={e => setEmailOrUsername(e.target.value)}
        />
      </div>
      <div>
        <label htmlFor="password">Password:</label>
        <input
          type="password"
          id="password"
          value={password}
          onChange={e => setPassword(e.target.value)}
        />
      </div>
      <button
        type="submit"
        onClick={loginUser}
      >
        Login
      </button>
    </form>
  );
};

const mapStateToProps = state => ({
  error: state.error,
});

export default connect(mapStateToProps, { login })(LoginPage);
