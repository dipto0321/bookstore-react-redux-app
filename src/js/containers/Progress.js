import React from 'react';
import CircularProgressbar from 'react-circular-progressbar';

const Progress = () => {
  const percentage = 66;
  return (
    <div className="progress-body">
      <div className="progress-bar">
        <CircularProgressbar percentage={percentage} />
      </div>
      <div className="progress-percentage">
        <p className="percent-complete">{`${percentage}%`}</p>
        <p className="completed">Completed</p>
      </div>
    </div>
  );
};

export default Progress;
