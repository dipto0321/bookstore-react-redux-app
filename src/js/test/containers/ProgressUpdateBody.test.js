import React from 'react';
import { shallow } from 'enzyme';
import { ProgressUpdateBody } from '../../containers/ProgressUpdateBody';

describe('ProgressUpdateBody', () => {
  let wrapper;

  beforeEach(() => {
    const chaptersArray = [1, 2, 3, 4, 5];
    wrapper = shallow(
      <ProgressUpdateBody
        chaptersArray={chaptersArray}
        id="someId"
        title="Some Title"
      />,
    );
  });

  test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should handle change in chapter', () => {
    wrapper.find('select').prop('onChange')({
      target: {
        name: 'current',
        value: '3',
      },
    });

    expect(wrapper.state('current')).toBe('3');
  });
});
