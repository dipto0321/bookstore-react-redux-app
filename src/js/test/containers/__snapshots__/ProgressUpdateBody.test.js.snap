// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressUpdateBody should render correctly 1`] = `
<div>
  <h1>
    Some Title
  </h1>
  <strong>
    Now reading chapter: 
  </strong>
  <select
    name="current"
    onChange={[Function]}
    value="1"
  >
    <option
      id="someId 1
                "
      key="someId-1
                "
      value={1}
    >
      1
    </option>
    <option
      id="someId 2
                "
      key="someId-2
                "
      value={2}
    >
      2
    </option>
    <option
      id="someId 3
                "
      key="someId-3
                "
      value={3}
    >
      3
    </option>
    <option
      id="someId 4
                "
      key="someId-4
                "
      value={4}
    >
      4
    </option>
    <option
      id="someId 5
                "
      key="someId-5
                "
      value={5}
    >
      5
    </option>
  </select>
</div>
`;
